File: SeniorProject-Lily/src/Class.java
public class Class {

    String name;
    String subject="";
    String courseNum ="";
    String level ="";
    String title = "";
    String grade = "";
    double creditHours = 0.0;
    double finalCredit = 0.0;
    boolean isMajorClass = false;
    boolean isCompElective = false;
    boolean isRequired = false;
    boolean isTransferCredit= false;

    public void setName(){
        name = subject + " " + courseNum;
    }


    @Override
    public String toString(){
        return subject +" "+courseNum+" "+ level+" "+title+" "+grade+ " "+ creditHours
                +" "+finalCredit+"\n "+"Major:"+isMajorClass+" Elective:"+isCompElective
                +" Required:"+isRequired+" Transfer Credit: "+isTransferCredit+"\n";
    }
}

File: Code/js/Class.js
class Class {

    Class(){
        this.name = "NONE";
        this.subject = "comp";
        this.courseNum = "163";
        this.level = "";
        this.title = "";
        this.grade = "";
        this.creditHours = 0.0;
        this.finalCredit = 0.0;
        this.isMajorClass = false;
        this.isCompElective = false;
        this.isRequired = false;
        this.isTransferCredit = false;
    }

    setName() {
        this.name = this.subject + " " + this.courseNum;
    }
    getName() {
        return classObj.name;
    }
}